core/src/main/java/nu/yona/server/subscriptions/entities/PrivateUserProperties.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.entities;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.persistence.Convert;
+import javax.persistence.MappedSuperclass;
+
+import nu.yona.server.crypto.seckey.StringFieldEncryptor;
+import nu.yona.server.crypto.seckey.UUIDFieldEncryptor;
+
+@MappedSuperclass
+public abstract class PrivateUserProperties extends EntityWithUuidAndTouchVersion
+{
+
+	@Convert(converter = StringFieldEncryptor.class)
+	private String firstName;
+
+	@Convert(converter = StringFieldEncryptor.class)
+	private String lastName;
+
+	@Convert(converter = StringFieldEncryptor.class)
+	private String nickname;
+
+	@Convert(converter = UUIDFieldEncryptor.class)
+	private UUID userPhotoId;
+
+	// Default constructor is required for JPA
+	public PrivateUserProperties()
+	{
+		super(null);
+	}
+
+	public PrivateUserProperties(UUID id, String firstName, String lastName, String nickname)
+	{
+		super(id);
+		this.firstName = Objects.requireNonNull(firstName);
+		this.lastName = Objects.requireNonNull(lastName);
+		this.nickname = nickname;
```

Comment:
Can `nickname` really be `null`? I think it shouldn't be, or should be `Optional`.

Reply:
It can indeed be null, in case the user is created on buddy request. Making it ``Optional`` makes sense, but it's not very usual in our entities.

I'll see how big the impact is.

Reply:
This would imply a major extension of this PR, well beyond its scope.
If we want this, we should do a structural refactoring to use ``Optional`` more widely.

Reply:
OK, no problem to skip it and do it later.